cannon/mipsevm/mips.go

```diff
@@ -179,6 +179,47 @@ func (m *InstrumentedState) handleSyscall() error {
 	return nil
 }
 
+func (m *InstrumentedState) pushStack(target uint32) {
+	if !m.debugEnabled {
+		return
+	}
+	m.debug.stack = append(m.debug.stack, target)
+	m.debug.caller = append(m.debug.caller, m.state.PC)
+}
+
+func (m *InstrumentedState) popStack() {
+	if !m.debugEnabled {
+		return
+	}
+	if len(m.debug.stack) != 0 {
+		fn := m.debug.meta.LookupSymbol(m.state.PC)
+		topFn := m.debug.meta.LookupSymbol(m.debug.stack[len(m.debug.stack)-1])
+		if fn != topFn {
```

Comment:
If there is no metadata file, `fn` and `topFn` will always be equal, right?  Do we need any special handling if we can't lookup symbols?

Reply:
A metadata file is required to use the debugger. I'll add an error return value to `InitDebug` to guard against missing metadata.